crates/adapters/src/server/prometheus.rs

```diff
@@ -15,22 +15,61 @@ pub(crate) struct PrometheusMetrics {
     pipeline_name: String,
     input_metrics: BTreeMap<EndpointId, InputMetrics>,
     output_metrics: BTreeMap<EndpointId, OutputMetrics>,
+    global_metrics: GlobalMetrics,
     handle: PrometheusHandle,
 }
 
 impl PrometheusMetrics {
     pub(crate) fn new(controller: &Controller) -> AnyResult<Self> {
-        let pipeline_name = controller
-            .status()
+        let status = controller.status();
+
+        let pipeline_name = status
             .pipeline_config
             .name
             .as_ref()
             .map_or_else(|| "unnamed".to_string(), |n| n.clone());
+
+        let global_metrics = GlobalMetrics {
+            cpu_msecs: GaugeBuilder::new("cpu_msecs", pipeline_name.clone())
+                .with_description("cpu time used by the pipeline process in milliseconds")
+                .with_unit(metrics::Unit::Milliseconds)
+                .build(),
+            rss_bytes: GaugeBuilder::new("rss_bytes", pipeline_name.clone())
+                .with_description("resident set size of the pipeline process in bytes")
+                .with_unit(metrics::Unit::Bytes)
+                .build(),
+            buffered_input_records: GaugeBuilder::new(
+                "buffered_input_records",
+                pipeline_name.clone(),
+            )
+            .with_description("total number of records currently buffered by all endpoints")
+            .with_unit(metrics::Unit::Count)
+            .build(),
+            total_input_records: GaugeBuilder::new("total_input_records", pipeline_name.clone())
+                .with_description("total number of input records received from all endpoints")
```

Comment:
```suggestion
                .with_description("total number of input records received from all connectors")
```

We are not yet consistent about it, but I think we should talk about connectors now, not endpoints.